Lib/datetime.py

```diff
@@ -1375,9 +1442,11 @@ def __hash__(self):
                 assert not m % timedelta(minutes=1), "whole minute"
                 m //= timedelta(minutes=1)
                 if 0 <= h < 24:
-                    self._hashcode = hash(time(h, m, self.second, self.microsecond))
```

Comment:
Again, this was an unnecessary style change. But in this case, it seems like there *should* have been a change here; `self.submicrosecond` should surely be part of the hash?

Reply:
fixed